common/src/main/nl/jft/common/util/elo/loader/FileEloExpectationLoader.java

```diff
@@ -3,49 +3,53 @@
 import nl.jft.common.util.elo.EloExpectation;
 import nl.jft.common.util.elo.EloExpectationLoader;
 
-import java.io.File;
+import java.io.BufferedReader;
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Scanner;
+import java.util.Objects;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
- * The {@code FileEloExpectationLoader} class is an implementation of the {@code EloExpectationLoader} interface that
- * retrieves a list of {@link EloExpectation EloExpectations} from the given path.
+ * The {@code FileEloExpectationLoader} class is an implementation of the {@link EloExpectationLoader} interface that
+ * retrieves a list of {@link EloExpectation EloExpectations} from a given path.
  *
  * @author Oscar de Leeuw
+ * @author Lesley
  */
 public class FileEloExpectationLoader implements EloExpectationLoader {
+
     private static final Logger LOGGER = Logger.getLogger(FileEloExpectationLoader.class.getName());
 
-    private String path;
+    private final Path path;
 
     /**
      * Creates a new {@code FileEloExpectationLoader}.
      *
-     * @param path The path to the {@code File} with the data.
+     * @param path The path to the {@code File} with the data, should not be {@code null}.
      */
-    public FileEloExpectationLoader(String path) {
-        this.path = path;
+    public FileEloExpectationLoader(Path path) {
+        this.path = Objects.requireNonNull(path);
     }
 
     @Override
     public List<EloExpectation> load() {
-        File file = new File(path);
-
-        List<EloExpectation> ret = new ArrayList<>();
+        List<EloExpectation> list = new ArrayList<>();
 
-        try (Scanner scanner = new Scanner(new FileInputStream(file))) {
-            while (scanner.hasNextLine()) {
-                ret.add(new EloExpectation(scanner.nextLine()));
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(path.toFile())))) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "FileInputStream". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://92.110.24.119:9000/coding_rules#rule_key=squid%3AS2095)